src/main/java/edu/cornell/kfs/coa/document/CuAccountGlobalMaintainableImpl.java

```diff
@@ -42,6 +47,13 @@ public List<MaintenanceLock> generateMaintenanceLocks() {
                 .filter(account -> isClosingAccount(account, oldAccountClosedStatuses))
                 .flatMap(this::generateTrickleDownMaintenanceLocks);
         
+        Cache cache = MaintenanceUtils.getBlockingCache();
+        String cacheKey = MaintenanceUtils.buildLockingDocumentCacheKey(getDocumentNumber());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("generateMaintenanceLocks, clear cache id: " + cacheKey);
+        }
+        cache.evictIfPresent(cacheKey);
```

Comment:
I'm wondering if the cache entry also needs to be cleared when the Account Global adds or removes collection lines, if this current mechanism is not enough to perform such clearing. If such an addition is needed, perhaps the `processAfterPost()` method could potentially be used for such entry clearing when it detects an add-line or remove-line operation, both single-valued and/or multi-valued.

Reply:
I am confused at what you're saying here.  Should this clear cache logic be moved to processAfterPost()?  If so, should this logic be moved to processAfterPost() in the other maintainable classes?

Reply:
If the user edits the Account Global doc between submission attempts, that may affect what downstream objects need maintenance locks during routing. This spot will at least clear the cache when the locks are being persisted to the database, but if this spot doesn't affect adding/removing accounts from the global doc, then appropriate cache clearing might need to be added to `processAfterPost()` or a similar hook.

The other maintainables don't need to override the `processAfterPost()` logic, since they only need to clear the cache when they store or delete locks that may affect Account Global. However, after considering this a little further, the other maintainables may want to move the cache clearing from their `generateMaintenanceLocks()` methods to their `doRouteStatusChange()` methods, so that the cache gets cleared when the objects start or finish their routing.

Similarly, it might make sense for the Account Global maintainable to move its single-entry clearing from `generateMaintenanceLocks()` to `doRouteStatusChange()` for clarity, unless that messes up the new caching feature.